go/kbfs/libkbfs/favorites.go

```diff
@@ -630,10 +672,22 @@ func (f *Favorites) RefreshCache(ctx context.Context) {
 		ctx:     context.Background(),
 	}
 	f.wg.Add(1)
+
+	if async {
+		favResult, err := f.config.KBPKI().FavoriteList(ctx)
+		if err != nil {
+			f.log.CDebugf(ctx, "Failed to refresh cached Favorites: %+v", err)
+			f.wg.Done()
+			<-f.refreshWaiting
```

Comment:
Won't this deadlock with the `<-f.refreshWaiting` here?
https://github.com/keybase/client/pull/16733/files#diff-5919763ea12fd1848e70a4ca1d6ce411R328

Reply:
The only place a request with `refresh: true` is ever created is on line 670, right above here. That request would only get to the processing logic if the results had already been retrieved in this function, or if `async` were false.  In either case, we won't get to the channel drain on line 681, but we will get to either the drain on line 690 or the one in the processing logic.

Reply:
OK please comment with a quick explanation.

Reply:
done.